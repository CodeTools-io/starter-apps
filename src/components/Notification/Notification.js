import React, { useEffect } from 'react';

import styles from './Notification.module.css';

export default function Notification({
  children,
  message,
  expiration,
  onExpire,
  ...props
}) {
  useEffect(() => {
    if (expiration && onExpire) {
      const timeoutId = setTimeout(onExpire, expiration);
      return () => {
        clearTimeout(timeoutId);
      };
    }
  }, [expiration, onExpire]);
  return (
    <div className={styles.Notification} {...props}>
      {message}
      {children}
    </div>
  );
}
