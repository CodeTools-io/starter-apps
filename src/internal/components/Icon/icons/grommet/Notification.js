import React from 'react';
import { Notification } from 'grommet-icons';

export default function NotificationIcon({ ...props }) {
  return <Notification {...props} />;
}
