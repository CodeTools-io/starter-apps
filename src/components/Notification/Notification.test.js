import React from 'react';
import Notification from './Notification';

jest.useFakeTimers();

describe('Notification', () => {
  test('can render children', () => {
    const { container } = renderComponent(
      <Notification>
        <p>Content for the Notification</p>
      </Notification>
    );

    expect(container).toMatchSnapshot();
  });
  test('can render message', () => {
    const { container } = renderComponent(
      <Notification message={'Content for the Notification'} />
    );

    expect(container).toMatchSnapshot();
  });

  test('can handle expirations', () => {
    const callback = jest.fn();
    renderComponent(
      <Notification
        message={'Content for the Notification'}
        expiration={1000}
        onExpire={callback}
      />
    );

    expect(callback).not.toBeCalled();
    jest.runAllTimers();
    expect(callback).toBeCalled();
  });
});
