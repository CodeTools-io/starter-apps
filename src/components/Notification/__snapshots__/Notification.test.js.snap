// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification can render children 1`] = `
<div>
  <div
    class="Notification"
  >
    <p>
      Content for the Notification
    </p>
  </div>
</div>
`;

exports[`Notification can render message 1`] = `
<div>
  <div
    class="Notification"
  >
    Content for the Notification
  </div>
</div>
`;
